From 4e2228f076b7d36da4df864cbc71946095adaecf Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Mon, 10 Feb 2020 14:11:24 +0200
Subject: [PATCH] [SW-8229] goya2: redirect flash offset on pldm to start

Because pldm is emulating a 8MB flash only we dont have access
to the whole 16MB flash like in the fpga or asic variant.
So we have to use on pldm a differant flash offset

Change-Id: I11caecb00201b6bd2c1683178881f6ab3db1bc16
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/Makefile b/Makefile
index 330c3e5..7bb2bf5 100644
--- a/Makefile
+++ b/Makefile
@@ -25,6 +25,11 @@
 ARCBOOT_CFLAGS += -I$(dir $(BR2_EXTERNAL))/embedded/specs
 ARCBOOT_CFLAGS += -DBTLVER=$(ARCBOOT_VER)
 ARCBOOT_CFLAGS += -DBTLID=0x30
+ifneq ($(HL_TARGET),pldm)
+	ARCBOOT_CFLAGS += -DFLASH_START_OFFSET=0xC00000
+else
+	ARCBOOT_CFLAGS += -DFLASH_START_OFFSET=0x0
+endif
 
 LDFLAGS = -flto
 LDFLAGS += -L $(LIBDIR) -lsb -lellipsys -L $(HL_DEV)
diff --git a/common/spi_flash.h b/common/spi_flash.h
index 3a75f35..c36509c 100644
--- a/common/spi_flash.h
+++ b/common/spi_flash.h
@@ -40,7 +40,7 @@
 #define FLASH_OTP_FW_VERSION_OFFSET	160
 #define FLASH_REVOKE_BITS_PER_DWORD	4
 
-#define FLASH_ARC_BASE_ADDRESS		0xC00000
+#define FLASH_ARC_BASE_ADDRESS		FLASH_START_OFFSET
 #define FLASH_PRIMARY_OFFSET		FLASH_ARC_BASE_ADDRESS
 #define FLASH_SECONDARY_OFFSET		(FLASH_PRIMARY_OFFSET + \
 						sizeof(uint32_t))
